app/models/taxon_range.rb

```diff
@@ -1,9 +1,14 @@
 class TaxonRange < ApplicationRecord
-  
   belongs_to :taxon
   belongs_to :source
+  belongs_to :user
+  has_updater
   has_many :listed_taxa, :dependent => :nullify
-  
+
+  validates_uniqueness_of :taxon_id, :message => "taxon range already exists"
```

Comment:
This results in an error like "Taxon taxon range already exists." I would leave the message blank, or change it to "already has a range" so the message reads "Taxon already has a range" instead. Actually, this text should be globalized, but I'm not actually sure how to do that properly off the top of my head.